### PR TITLE
Implement MainMenu::DrawOptionsMenu

### PR DESCRIPTION
**Pull request by Renzo904**
Lets just say i'm not proud of this abomination i just made

Note:All the `foo = foo + 1` cant be refactored to `foo++` because otherwise the asm changes from `inc foo`(original) to `add foo, 1`

### CODE REVIEW

File: src/MainMenu.cpp [hunk 1]
```diff
@@ -1769,4 +1769,256 @@ ZunResult MainMenu::ChoosePracticeLevel()
...
+
+            case CURSOR_OPTIONS_POS_MUSICMODE:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                g_Supervisor.StopAudio();
+                if (g_Supervisor.cfg.musicMode <= OFF)
+                {
+                    g_Supervisor.cfg.musicMode = MIDI + 1;
+                }
+                g_Supervisor.cfg.musicMode = g_Supervisor.cfg.musicMode - 1;
```
Comment (roblabla) [inline src/MainMenu.cpp:1867]:
```suggestion
                g_Supervisor.cfg.musicMode -= 1;
```

While you can't use --, you can use -= I believe.

File: src/MainMenu.cpp [hunk 2]
```diff
@@ -1769,4 +1769,256 @@ ZunResult MainMenu::ChoosePracticeLevel()
 }
 #pragma optimize("", on)
 
+#pragma optimize("s", on)
+#pragma var_order(i, optionsVm)
+u32 MainMenu::DrawOptionsMenu()
```
Comment (roblabla) [inline src/MainMenu.cpp:1774]: I just realized this should really be `OnUpdateOptionsMenu`

File: src/MainMenu.cpp [hunk 3]
```diff
@@ -1769,4 +1769,256 @@ ZunResult MainMenu::ChoosePracticeLevel()
...
+    AnmVm *optionsVm;
+    i32 i;
+
+    MoveCursor(this, 9);
+    optionsVm = &this->vm[8];
+    for (i = 0; i < 9; i++)
+    {
+        if ((i >= 5) && (i <= 7))
+        {
+            this->setSavedCursorPosition(&this->vm[i + 0x43], i, i, this->cursor);
```
Comment (roblabla) [inline src/MainMenu.cpp:1786]:
```suggestion
            this->SetSavedCursorPosition(&this->vm[i + 0x43], i, i, this->cursor);
```

File: src/MainMenu.cpp [hunk 4]
```diff
@@ -1769,4 +1769,256 @@ ZunResult MainMenu::ChoosePracticeLevel()
...
+    {
+        this->setSavedCursorPosition(optionsVm, CURSOR_OPTIONS_POS_PLAYSOUNDS, i, g_Supervisor.cfg.playSounds);
+    }
+    optionsVm = &this->vm[0x4d];
+
+    for (i = 0; i < 3; i++, optionsVm++)
+    {
+        this->setSavedCursorPosition(optionsVm, CURSOR_OPTIONS_POS_MUSICMODE, i, g_Supervisor.cfg.musicMode);
+    }
+    optionsVm = &this->vm[0x4b];
```
Comment (roblabla) [inline src/MainMenu.cpp:1818]:
```suggestion
    optionsVm = &this->vm[75];
```

File: src/MainMenu.cpp [hunk 5]
```diff
@@ -1769,4 +1769,256 @@ ZunResult MainMenu::ChoosePracticeLevel()
...
+                if (g_Supervisor.cfg.colorMode16bit >= 2)
+                {
+                    g_Supervisor.cfg.colorMode16bit = 0;
+                }
+                break;
+            case CURSOR_OPTIONS_POS_MUSICMODE:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                g_Supervisor.StopAudio();
+                g_Supervisor.cfg.musicMode = g_Supervisor.cfg.musicMode + WAV;
```
Comment (roblabla) [inline src/MainMenu.cpp:1933]:
```suggestion
                g_Supervisor.cfg.musicMode += 1;
```

It doesn't make much sense to use WAV here IMO. We're just moving by 1 through the enum.

**Review by roblabla — APPROVED** [review]
Looking good! Just a few nitpicks here and there and this is good to merge!

File: src/MainMenu.cpp (hunk 4 above)
Comment (Renzo904) [reply src/MainMenu.cpp:1818]: Should i use hex on the next if or not?
```diff 
- if (this->stateTimer >= 0x20)
+ if (this->stateTimer >= 32)
```

File: src/MainMenu.cpp (hunk 5 above)
Comment (Renzo904) [reply src/MainMenu.cpp:1933]: You're right, it doesn't, it does make sense with MIDI cause its checking if the cursor its at the end

File: src/MainMenu.cpp (hunk 4 above)
Comment (roblabla) [reply src/MainMenu.cpp]: Yeah, I think that makes more sense. stateTimer is a frame counter, I don't think using hex for those make much sense.

**Pull request merged**